src/ctapipe/image/muon/ring_fitter.py

```
@@ -22,7 +22,26 @@ def taubin(fov_lon, fov_lat, weights, mask):
     return taubin_circle_fit(fov_lon, fov_lat, mask)
 
 
-FIT_METHOD_BY_NAME = {m.__name__: m for m in [kundu_chaudhuri, taubin]}
+def kundu_chaudhuri_taubin(fov_lon, fov_lat, weights, mask):
+    """taubin_circle_fit with fov_lon, fov_lat, weights, mask interface
+    with initial parameters provided by kundu_chaudhuri"""
+    taubin_r_initial, xc_initial, yc_initial, _, _, _ = kundu_chaudhuri_circle_fit(
```

I asked in another review if there would be a reason to switch off the error computation and instead of replying, you removed the switch.

But here the errors are ignored, so there was actually a reason.

If the error computation has a sizeable impact on the performance, then it should be skipped. 
It could also be used though for the initial step width in the Minuit fit I think.

---

The performance is not degraded, and the errors can be used—and will be used—later in this step. So this was a good remark.

---

I was talking about runtime. The uncertainty computation perform many more operations than the actual fit function.

---

yes, but it is  ok for the CPU. In any case we will use it this error calculation.

---

How did you make this decision? How can performing unneeded computations "be ok for the CPU"?

Please either use the values, or do not compute them.

---

@maxnoe .... Ok i putting back the nans.